core/src/main/scala/org/bykn/bosatsu/Matchless.scala

```diff
@@ -173,7 +173,20 @@ object Matchless {
         l.nonEmpty || hasSideEffect(b)
     }
 
-  case class If(cond: BoolExpr, thenExpr: Expr, elseExpr: Expr) extends Expr
+  case class If(cond: BoolExpr, thenExpr: Expr, elseExpr: Expr) extends Expr {
+    def flatten: (NonEmptyList[(BoolExpr, Expr)], Expr) = {
+      def combine(expr: Expr): (List[(BoolExpr, Expr)], Expr) =
+        expr match {
+          case If(c1, t1, e1) =>
+            val (ifs, e2) = combine(e1)
+            (((c1, t1)) :: ifs, e2)
```

Comment:
previously this code was in PythonGen, but it really only depends on Matchless.Expr. The order was being reversed by appending to the tail: `ifs :+ ((c1, t1))` which for some branches changes semantics (where Matchless has leveraged prior information to do less work).